src/vsphere_cpi/lib/cloud/vsphere/resources/network.rb

```diff
@@ -0,0 +1,93 @@
+require 'cloud/vsphere/resources'
+
+module VSphereCloud
+  module Resources
+    class Network
+      include VimSdk
+      include ObjectStringifier
+      stringify_with :name, :mob, :bosh_name
+
+      attr_reader :mob, :client, :bosh_name, :vim_name
+
+      def self.NetworkBuilder(bosh_name, mob, client)
+        if mob.is_a?(VimSdk::Vim::Dvs::DistributedVirtualPortgroup)
+          DistributedVirtualPortGroupNetwork.new(bosh_name, mob, client)
+        elsif mob.is_a?(VimSdk::Vim::OpaqueNetwork)
+          OpaqueNetwork.new(bosh_name, mob, client)
+        else
+          Network.new(bosh_name, mob, client)
+        end
+      end
+
+      def initialize(bosh_name, mob, client)
+        @bosh_name = bosh_name
+        @mob = mob
+        @vim_name = @mob.name
```

Comment:
Seems like this should be a method rather than an instance variable and `attr_reader`.

Reply:
Done